server/lib/data.ts

```diff
@@ -45,6 +51,53 @@ export function diffDBEntries<T extends Model>(
   return [toCreate, toRemove, toUpdate];
 }
 
+type RichDBEntriesDiff<BaseModel extends Model, CreateOrUpdateAttributes = Record<string, unknown>> = {
+  toCreate: Array<CreateOrUpdateAttributes>;
+  toRemove: Array<BaseModel>;
+  toUpdate: Array<{ model: BaseModel; newValues: CreateOrUpdateAttributes }>;
+};
+
+/**
+ * Diff two lists of DB objects and returns which ones where created, removed and updated.
+ * Useful for places where we update an attribute by providing a list.
+ *
+ * @param {Array} oldEntries: Existing entries
+ * @param {Array} newEntries: New entries to update with
+ * @param {Array} diffedFields: The fields used to compare objects for ``
+ * @returns [newEntries, removedEntries, updatedEntries]
+ */
+export function richDiffDBEntries<BaseModel extends Model, CreateOrUpdateAttributes = Record<string, unknown>>(
```

Comment:
This function is a copy of `diffDBEntries` with the following adjustments:
- It returns an object with named properties, which is more evident than a tuple
- Inputs can be better typed than `Record` thanks to  the `CreateOrUpdateAttributes` template param
- Update values are passed alongside the original model, which allows for better recording before/after values (i.e., in activities) and can simplify the model update in some cases

Converting existing code would be easy, but I preferred to keep it for a separate PR.